### PR TITLE
istioctl: add command to generate/apply waypoints more easily 

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/43279

### CODE REVIEW

File: istioctl/cmd/waypoint.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		if name == "" {
+			name = "namespace"
+		}
+		gw := gateway.Gateway{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       gvk.KubernetesGateway.Kind,
+				APIVersion: gvk.KubernetesGateway.GroupVersion(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
```
Comment (hanxiaop) [inline istioctl/cmd/waypoint.go:50]: This is missing the `--waypoint` suffix, which is in https://github.com/istio/istio/issues/43279 issue's template.

Comment (howardjohn) [reply istioctl/cmd/waypoint.go:50]: Issue is wrong, the Deployment name gets `-istio-waypoint` added already so putting `waypoint` twice is redundant

**Review by linsun — APPROVED** [review]
LGTM, thanks for adding a test for it as well!

**Pull request merged**